### ISSUE TITLE
AWS::ECS::CapacityProvider-ManagedScaling-InstanceWarmupPeriod support in Cloudformation

### ISSUE THREAD

**Issue by damongabrielle** [aws-cloudformation/cloudformation-coverage-roadmap#854]
## 2. Scope of request 
While setting the InstanceWarmupPeriod for CapacityProviders::Managed scaling is supported from the aws CLI and SDK, I would really like to see support for setting this attribute in the cloudformation template as well. 

## 3. Expected behavior
This would allow setting the warmup period when a codepipeline script executes my cloudformation. Right now I'm setting this value from the CLI after my capacity provider has been created in my cloudformation stack. I don't have this automated either, it's just a manual step right now so it's sort of risky. Should I ever delete the stack and run cloudformation to recreate it again,  I need to remember to execute the cli command again to override the default value.

Side note: In my application, which uses autoscaling to provide testing resources, 300 seconds feels like an eternity to wait for an ECS when a cluster is "cold starting" as well as after an autoscaling event ocurrs (when the cluster is also out of resources).  I'm not asking that you reconsider the default value for this issue but I do wonder if many share this sentiment and thus are interested in overriding the default at the time they create / update the capacity provider resource.
Basically I'm looking to support this:

### 4. Sample Cloudformation:

```
"ECSCapacityProvider": {
      "Type": "AWS::ECS::CapacityProvider",
      "Properties": {
        "Name": "MyCapacityProvider"
        "AutoScalingGroupProvider": {
          "AutoScalingGroupArn": "autoscalingGroupShortName",
          "ManagedScaling": {
            "MaximumScalingStepSize": 10000,
            "MinimumScalingStepSize": 1,
            "Status": "ENABLED",
            "TargetCapacity": 100,
            "InstanceWarmupPeriod": 20 <========= THIS IS WHAT I'D LIKE TO BE ABLE TO SET
          },
          "ManagedTerminationProtection": "ENABLED"
        }
      },
      "DependsOn" : "ECSAutoScalingGroup"
    }
```

**Comment by damongabrielle** [comment]
Just want to clarify that I'm currently setting the InstanceWarmupPeriod successfully using the available aws CLI command `aws ecs update-capacity-provider --cli-input-json file://myCapacityProvider.json`. I guess I thought maybe that meant the plumbing was already there to support CFN @drs5ma . I wasn't sure what you meant by `also impacted` so I thought I'd mention that.

**Comment by kddejong** [comment]
I believe this has shipped.  Can you validate this is what you are looking for?

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-capacityprovider-managedscaling.html
